utils/cautils/token_generator.go

```diff
@@ -161,32 +162,60 @@ func generateK8sSAToken(ctx *cli.Context, p *provisioner.K8sSA) (string, error)
 func generateX5CToken(ctx *cli.Context, p *provisioner.X5C, tokType int, tokAttrs tokenAttrs) (string, error) {
 	x5cCertFile := ctx.String("x5c-cert")
 	x5cKeyFile := ctx.String("x5c-key")
+	x5cChainFiles := ctx.StringSlice("x5c-chain")
+	kmsURI := ctx.String("kms")
+
 	if x5cCertFile == "" {
 		return "", errs.RequiredWithProvisionerTypeFlag(ctx, "X5C", "x5c-cert")
 	}
 	if x5cKeyFile == "" {
 		return "", errs.RequiredWithProvisionerTypeFlag(ctx, "X5C", "x5c-key")
 	}
 
-	// Get private key from given key file
-	var opts []jose.Option
-	if passOpt := getProvisionerPasswordOption(ctx); passOpt != nil {
+	var jwk *jose.JSONWebKey
+	var err error
+
+	var opts []pemutil.Options
+	if passOpt := getProvisionerPasswordPEMOption(ctx); passOpt != nil {
 		opts = append(opts, passOpt)
 	}
-	jwk, err := jose.ReadKey(x5cKeyFile, opts...)
+
+	var kmsSigner crypto.Signer
+	kmsSigner, err = cryptoutil.CreateSigner(ctx.String("kms"), x5cKeyFile, opts...)
 	if err != nil {
 		return "", err
 	}
 
+	joseSigner := jose.NewOpaqueSigner(&kmsSigner)
```

Comment:
This should be:
```suggestion
	joseSigner := jose.NewOpaqueSigner(kmsSigner)
```

It's not a good idea to use a pointer to an interface. I fixed that in a PR.